tests/test_select.py

```diff
@@ -99,6 +100,23 @@ def test_select_default_throw_on_missing(
         OmegaConf.select(cfg, key, default=default, throw_on_missing=True)
 
 
+@pytest.mark.parametrize(
+    "cfg, key",
+    [
+        pytest.param(["???"], "0", id="missing"),
+    ],
+)
+def test_select_default_throw_on_missing_list(
```

Comment:
And this function should also test dicts.

Reply:
This was also part of #564

Reply:
Resolving since this is not part of this PR anymore